app/presenters/hyrax/displays_iiif.rb

```diff
@@ -0,0 +1,58 @@
+
+module Hyrax
+  # This gets mixed into a work presenter in order to handle
+  # iiif related methods
+  module DisplaysIIIF
+    extend ActiveSupport::Concern
+
+    IIIF_DEFAULT_VERSION = 2
+    IIIF_DEFAULT_MANIFEST_MIME = "application/json;profile=http://iiif.io/api/presentation/#{IIIF_DEFAULT_VERSION}/context.json".freeze
+
+    # @return [Boolean] render a IIIF viewer
+    def iiif_viewer?
+      representative_id.present? &&
+        representative_presenter.present? &&
+        (representative_presenter.video? ||
+         representative_presenter.audio? ||
+         (representative_presenter.image? && Hyrax.config.iiif_image_server?))
+    end
+
+    alias universal_viewer? iiif_viewer?
+    deprecation_deprecate universal_viewer?: "use iiif_viewer? instead"
+
+    def iiif_viewer
+      if representative_presenter.video? || representative_presenter.audio?
+        :avalon
+      elsif representative_presenter.image?
+        :universal_viewer
+      end
+    end
+
+    # @return the highest IIIF version (as an integer) specified in the Accept request header, or the default version if none specified
+    def iiif_version
+      accept = parse_accept(request.headers['Accept'])
```

Comment:
The fix for the bug was to parse the accept header every time and have the `parse_accept` method return `null` if the accept header is blank or doesn't match the regex.